frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +86,28 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // saveDocumentBeforeCheck needed in order to update the state of the .git/machete VirtualFile before actual check
```

Comment:
nit: I do not like method/class references from the usual comments. it is another thing to remember while refactoring. I would rather see this comment as javadoc in the place where the method is declared

Reply:
Changed the content of this comment. In fact, this is not a comment which tells about the method itself, so it should not be a javadoc. This is just an explanation, **why** we need to update the state of the .git/machete VirtualFile before performing the actual check within the `processMacheteGeneratedEntry` logic.